src/orchestrator/chat_completions_detection.rs

```diff
@@ -147,31 +173,129 @@ impl Orchestrator {
                         .unwrap()
                         .as_secs() as i64,
                     detections: Some(ChatDetections {
-                        input: input_detections,
+                        input: detections,
                         output: vec![],
                     }),
                     warnings: vec![OrchestratorWarning::new(
-                        InputWarningReason::UnsuitableInput,
+                        DetectionWarningReason::UnsuitableInput,
                         UNSUITABLE_INPUT_MESSAGE,
                     )],
                     ..Default::default()
                 })))
             } else {
-                let model_id = task.request.model.clone();
                 let client = ctx
                     .clients
                     .get_as::<OpenAiClient>("chat_generation")
                     .expect("chat_generation client not found");
                 let mut chat_request = task.request;
+                let model_id = chat_request.model.clone();
                 // Remove detectors as chat completion server would reject extra parameter
                 chat_request.detectors = None;
-                client
-                    .chat_completions(chat_request, task.headers)
+                let headers = task.headers.clone();
```

Comment:
this clone can be avoided by directly passing `task.headers` ?

Reply:
I get an error if I try to do that since I'm using `task.headers` inside the closure on line `212`.

```
borrow of moved value: `task.headers`
move occurs because `task.headers` has type `HeaderMap`, which does not implement the `Copy`
chat_completions_detection.rs(196, 53): value moved here
chat_completions_detection.rs(212, 51): borrow occurs due to use in closure
borrow of moved value: `task.headers`
move occurs because `task.headers` has type `http::HeaderMap`, which does not implement the `Copy` 
chat_completions_detection.rs(196, 53): value moved here
chat_completions_detection.rs(212, 51): borrow occurs due to use in closure
```

A lot of the clones were me trying to appease the complier :sweat_smile: